pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/design/AvoidUnusedMethodRule.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang.apex.rule.design;
+
+import static net.sourceforge.pmd.properties.constraints.NumericConstraints.positive;
+
+import java.util.List;
+
+import net.sourceforge.pmd.lang.apex.ast.ASTMethod;
+import net.sourceforge.pmd.lang.apex.ast.ApexRootNode;
+import net.sourceforge.pmd.lang.apex.rule.AbstractApexRule;
+import net.sourceforge.pmd.lang.ast.RootNode;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+import net.sourceforge.pmd.properties.PropertyFactory;
+
+import com.nawforce.common.diagnostics.Issue;
+
+public class AvoidUnusedMethodRule extends AbstractApexRule {
+
+    private static final PropertyDescriptor<Integer> PROBLEM_DEPTH_DESCRIPTOR
+            = PropertyFactory.intProperty("problemDepth")
+            .desc("The if statement depth reporting threshold")
+            .require(positive()).defaultValue(3).build();
+
+    public AvoidUnusedMethodRule() {
+        definePropertyDescriptor(PROBLEM_DEPTH_DESCRIPTOR);
+    }
+
+    @Override
+    public Object visit(ASTMethod node, Object data) {
+        // Check if any 'Unused' Issues align with this method
+        for (Issue issue: getIssues(node)) {
+            if (issue.diagnostic().category().value().equals("Unused")) {
```

Comment:
Is there a constant somewhere we could use instead of this hardcoded string?

Reply:
This is a 'case object' in Scala, I have swapped to use 'UNUSED_CATEGORY$.MODULE$ == issue.diagnostic().category()' but it's not very readable, I will introduce a proper Java enum for this in next release.